lib/LANraragi/Model/Tankoubon.pm

```diff
@@ -84,13 +101,59 @@ sub create_tankoubon ( $name, $tank_id ) {
 
     # Default values for new group
     # Score 0 will be reserved for the name of the tank
-    $redis->zadd( $tank_id, 0, redis_encode($name) );
+    $redis->zadd( $tank_id, $TANK_METADATA{"name"}, redis_encode("name_${name}") );
+    $redis->zadd( $tank_id, $TANK_METADATA{"summary"}, "summary_" );
+    $redis->zadd( $tank_id, $TANK_METADATA{"thumbhash"}, "thumbhash_" );
+    $redis->zadd( $tank_id, $TANK_METADATA{"tags"}, "tags_" );
+    $redis->zadd( $tank_id, $TANK_METADATA{"alias"}, "alias_" );
 
     $redis->quit;
 
     return $tank_id;
 }
 
+# update_metadata(tankoubonid, data)
+#   Updates the metadata in the Tankoubon.
+#   Returns 1 on success, 0 on failure alongside an error message.
+sub update_metadata( $tank_id, $data ) {
+
+    my $logger          = get_logger( "Tankoubon", "lanraragi" );
+    my $redis           = LANraragi::Model::Config->get_redis;
+    my $err             = "";
+    my $name            = $data->{"name"};
+    my $summary         = $data->{"summary"};
+    my $thumbhash       = $data->{"thumbhash"};
+    my $tags            = $data->{"tags"};
+    my $alias           = $data->{"alias"};
+
+    if ( $redis->exists($tank_id) ) {
+        $redis->zremrangebyscore( $tank_id, $TANK_METADATA{"name"}, $TANK_METADATA{"name"} );
```

Comment:
Those zremrangebyscore/zadd combos could probably be refactored into a reusable subroutine  

(And again, do we need the prefixes here for each value? The original name property didn't need one)

Reply:
Will check if I can refactor that, it's mostly a copy paste, so I don't think will be hard to do. For the prefixes check my previous comment, is an DB integrity protection.